### PR TITLE
Shapes exercise(s)

### PR DESCRIPTION
**Pull request by Mirabellensaft**

### CODE REVIEW

**Comment by Mirabellensaft** [comment]
There is a problem with /exercise-solutions/shapes-part-3 with implementing the trait `HasArea` for Circle. Can this issue be fixed? 
I have written instructions up until this part, so that is missing. 
Instead of the step-by-step solution I kept the old help section, were applicable and added some points.

**Comment by jonathanpallant** [comment]
`struct Circle` is not generic, so you implement `HasArea<T>` like this:

```rust
impl HasArea<f32> for Circle {

}
```

File: exercise-book/src/shapes.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+```console
+$ cargo new --lib shapes 
+$ cd shapes
+$ cargo run
+```
+
+### Creating a Type
+
+Each of your shape types (Square, Circle, etc) will need some fields (or
```
Comment (amanjeev) [inline exercise-book/src/shapes.md:88]:
```suggestion
Each of your shape types (`Square`, `Circle`, etc.) will need some fields (or
```

File: exercise-solutions/shapes-part-2/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+}
+
+impl HasArea for Circle {
+    fn area(&self) -> f32 {
+        Circle::area(self)
+    }
+}
+
+pub enum Shape {
```
Comment (jonathanpallant) [inline exercise-solutions/shapes-part-2/src/lib.rs:73]:
```suggestion
/// Allow us to handle any one of the given kinds of Shape within a single type.
pub enum Shape {
```

File: exercise-solutions/shapes-part-1/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        self.radius *= factor;
+    }
+
+    /// Takes ownership of the given [`Circle`] and returns the underlying value
+    pub fn destroy(self) -> f32 {
+        self.radius
+    }
+}
+
+pub enum Shape {
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply exercise-solutions/shapes-part-1/src/lib.rs:56]: I'll delete it.

File: exercise-solutions/shapes-part-3/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+impl<T> HasArea<T> for Circle<T>
+where
+    T: num::Num + Copy + From<f32>,
+{
+    fn area(&self) -> T {
+        Circle::area(self)
+    }
+}
+
+pub enum Shape<T> {
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply exercise-solutions/shapes-part-3/src/lib.rs:93]: Added doc comment.

**Comment by jonathanpallant** [comment]
OK, let's go!

**Pull request merged**